types/rdf-js/rdf-js-tests.ts

```diff
@@ -176,19 +177,27 @@ function test_dataset() {
     const stream1: Stream = <any> {};
     const stream2: Stream<QuadBnode> = <any> {};
 
-    const quadFilterIteratee1: QuadFilterIteratee = <any> {};
-    const quadFilterIteratee2: QuadFilterIteratee<QuadBnode> = <any> {};
-
-    const quadMapIteratee1: QuadMapIteratee = <any> {};
-    const quadMapIteratee2: QuadMapIteratee<QuadBnode> = <any> {};
-
-    const quadReduceIteratee1: QuadReduceIteratee = <any> {};
-    const quadReduceIteratee2: QuadReduceIteratee<string> = <any> {};
-    const quadReduceIteratee3: QuadReduceIteratee<any, QuadBnode> = <any> {};
-    const quadReduceIteratee4: QuadReduceIteratee<string, QuadBnode> = <any> {};
-
-    const quadRunIteratee1: QuadRunIteratee = <any> {};
-    const quadRunIteratee2: QuadRunIteratee<QuadBnode> = <any> {};
+    const quadFilterIteratee = (quad: Quad, dataset: Dataset) => {
```

Comment:
Could you add the explicit types for each iteratee?
It's good to have the tests as strongly typed as possible, to avoid unintentonal future breakages.

Reply:
I added the return types to callback signatures. You think that enough?

Reply:
While that's also good to have, I was referring to the types of the iteratees itself. So `const quadFilterIteratee: someType = ...`

Reply:
I'm not convinced.

The explicit type makes sense when verifying the return of tested call. In this case the type inference should be enough:

```ts
const quadFilterIteratee: (quad: Quad, dataset: Dataset) => boolean 
  = (quad: Quad, dataset: Dataset): boolean {}
```

I think this is unnecessarily redundant.

Reply:
Yes, it's redundant, but that's the whole point of these tests 🙂.
Things here should be as strongly typed as possible to avoid future regressions.

Reply:
Not really. This is not the subject under test. The actual interface methods are and there v everything is explicitly typed.

Here nothing is really implicit.

I'm out now so if you insist I can do b that v later. But seriously, would you also have me type

```js
const x: number = 5;
const y: string = "there is nothing implicit about this" 
```

?

Reply:
Alright, having re-read my own argumentation I realised that the implementation of those callback methods is just a distraction and clutter anyway. I replaced them with just variables typed to what I expect them to be.